File: maven/java-basic/src/test/java/com/example/FileReaderTest.java
package com.example;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class FileReaderTest {

    private final FileReader fileReader =  new FileReader();

    @Test
    public void shouldLoadFile() throws IOException {
        List<String> content = fileReader.readFile("test.txt");

        assertEquals(1, content.size());
        assertEquals("Hello World!", content.getFirst());
    }

    @Test
    public void shouldThrowExceptionWhenFileDoesNotExist() {
        assertThrows(IOException.class, () -> fileReader.readFile("invalid.txt"));
    }
}

File: _template/kotlin-basic/src/test/kotlin/com/example/FileReaderTest.kt
package com.example

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotNull
import kotlin.test.assertNull

class FileReaderTest {
    private val fileReader = FileReader()

    @Test
    fun shouldLoadFile() {
        val content = fileReader.readFile("test.txt")

        assertNotNull(content)
        assertEquals(1, content.size)
        assertEquals("Hello World!", content.single())
    }

    @Test
    fun shouldReturnNullWhenFileDoesNotExist() {
        val content = fileReader.readFile("invalid.txt")

        assertNull(content)
    }
}
